### ISSUE TITLE
Dataset creation crashes on GODataset k=5 graph

### ISSUE THREAD

**Issue by cgoliver** [BorgwardtLab/torch-pdb#45]
Using the latest version from PyPi

```
>>> from torch_pdb.datasets import GODataset
>>> GODataset(root="./data").to_graph(k=5).pyg() 
Converting proteins to graphs:  49%|████████████████████████████████████████████████████████████▎                                                              | 9448/19267 [00:27<00:26, 372.78it/s]Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/carlosgonzalezoliver/anaconda3/envs/biographs/lib/python3.9/site-packages/torch_pdb/datasets/dataset.py", line 292, in to_graph
    return GraphDataset(self.root, self.proteins, *args, **kwargs)
  File "/Users/carlosgonzalezoliver/anaconda3/envs/biographs/lib/python3.9/site-packages/torch_pdb/representations/graph.py", line 47, in __init__
    self.proteins = self.convert(proteins)
  File "/Users/carlosgonzalezoliver/anaconda3/envs/biographs/lib/python3.9/site-packages/torch_pdb/utils/io.py", line 129, in wrapper
    result = function(self, *args, **kwargs)
  File "/Users/carlosgonzalezoliver/anaconda3/envs/biographs/lib/python3.9/site-packages/torch_pdb/representations/graph.py", line 60, in convert
    return [self.protein2graph(p) for p in tqdm(proteins, desc='Converting proteins to graphs')]
  File "/Users/carlosgonzalezoliver/anaconda3/envs/biographs/lib/python3.9/site-packages/torch_pdb/representations/graph.py", line 60, in <listcomp>
    return [self.protein2graph(p) for p in tqdm(proteins, desc='Converting proteins to graphs')]
  File "/Users/carlosgonzalezoliver/anaconda3/envs/biographs/lib/python3.9/site-packages/torch_pdb/representations/graph.py", line 55, in protein2graph
    adj = kneighbors_graph(protein['coords'], n_neighbors=self.k, mode=mode)
  File "/Users/carlosgonzalezoliver/anaconda3/envs/biographs/lib/python3.9/site-packages/sklearn/neighbors/_graph.py", line 124, in kneighbors_graph
    return X.kneighbors_graph(X=query, n_neighbors=n_neighbors, mode=mode)
  File "/Users/carlosgonzalezoliver/anaconda3/envs/biographs/lib/python3.9/site-packages/sklearn/neighbors/_base.py", line 878, in kneighbors_graph
    A_ind = self.kneighbors(X, n_neighbors, return_distance=False)
  File "/Users/carlosgonzalezoliver/anaconda3/envs/biographs/lib/python3.9/site-packages/sklearn/neighbors/_base.py", line 724, in kneighbors
    raise ValueError(
ValueError: Expected n_neighbors <= n_samples,  but n_samples = 2, n_neighbors = 6
>>> 
```

**Comment by cgoliver** [comment]
looks like maybe some proteins are too small for a k=5 graph?

**Comment by timkucera** [comment]
looks like it, should be a one-liner `k = min(k,len(protein))`

**Comment by cgoliver** [comment]
on it

**Comment by cgoliver** [comment]
fixed in #47
